keps/sig-node/4369-allow-special-characters-environment-variable/README.md

```diff
@@ -0,0 +1,290 @@
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Allow all printable ASCII except '=', to be set as environment variables.
+
+## Motivation
+
+Kubernetes should not restrict which environment variable names can be used, because it has no way of knowing what the application may need, and people can't always choose their own variable names, which may limit the adoption of Kubernetes.
+
+### Goals
+
+* Allow users to set printable ASCII other than '=' as environment variables.
+
+### Non-Goals
+
+## Proposal
+
+* Implements relaxed validation at the top level validation method when validating API create requests.
+* Document rules for setting environment variables.
+
+### User Stories (Optional)
+
+#### Story 1
+
+I am a .NET Core development engineer, .Net Core applications are using ":" when working with application settings loaded from appsettings.json file. When running .net core app in containers typically overwrite this settings by specifying environmental variable.
+such as: 
+`"Logging": { "IncludeScopes": false, "LogLevel": { "Default": "Warning" } }`    
+override like this `-e Logging:LogLevel:Default=Debug`    
+
+### Risks and Mitigations
+
+Relaxed validation can break upgrade and rollback scenarios, but our use of feature gate to control whether it's enabled or not will make it a manageable risk, with the user having the autonomy to choose whether or not to enable it.
+
+## Design Details
+
+- A feature gate name `RelaxedValidationEnvironment` controlling the loosening of the envvar name validation, initially in alpha state and defaulting to false
+- Two sets of validation logic for envvar names:
+  - Current validation
+  - Relaxed validation
+- Everywhere we validate envvar names in API objects, plumbing a parameter whether we want the strict or relaxed validation
```

Comment:
Does this include PodTemplate and its embeddings?

Reply:
EnvVar is a field belonging to PodTemplate, we just need to verify the EnvVar name.